modules/04_PE_FE_parameters/iea2014/datainput.gms

```diff
@@ -13,6 +13,27 @@ $include "./modules/04_PE_FE_parameters/iea2014/input/f04_IO_input.cs4r"
 $offdelim
 /
 ;
+
+file diagnosis_f04_input_negatives;
+if( smin( (t,regi,entyPe,entySe,te), f04_IO_input(t,regi,entyPe,entySe,te)) < 0,
+  put diagnosis_f04_input_negatives;
+  put "f04_IO_input has the following negative values that are overwritten by 0 in 04/iea2014/datainput.gms" //;
+  put "te", @15, "regi", @20, "value"//;
+  loop(t,
+    loop(regi,
+      loop(pe2se(entyPe,entySe,te),
+        if( f04_IO_input(t,regi,entyPe,entySe,te)  lt 0,
+          put_utility "msg" / "**""** input data problem: f04_IO_input has negative values that are overwritten to still allow model solving." ;
+          put_utility "msg" / "Check input data. More details in the file diagnosis_f04_input_negatives";
+          put te.tl, @ 15, regi.tl, @20, f04_IO_input(t,regi,entyPe,entySe,te):10:8 /;
+        );
+      );
```

Comment:
You want the message to the .lst file in the loop?  How many negative values are there?

Reply:
no, right, I'll also take it outside. only the writing to the put file needs to be in the loop. 
Thanks!

(and how do you get notified of draft pull requests for which I  haven't even assigned a reviewer? :-)

Reply:
> and how do you get notified of draft pull requests for which I haven't even assigned a reviewer? 

Change your notification settings in the top right corner:

![image](https://user-images.githubusercontent.com/7226087/212947855-af4b6424-b49d-490b-b51f-ba41de141eea.png)

And enjoy the spam :slightly_smiling_face:

Reply:
Stasi-Mode.